### PR TITLE
smart_classroom: check action number

### PR DESCRIPTION
**Pull request by Wovchena**

### CODE REVIEW

**Comment by Wovchena** [comment]
@IRDonch feel free to retarget, if it should be `release` branch.

**Comment by Wovchena** [comment]
Jenkins please retry a build

File: demos/smart_classroom_demo/src/action_detector.cpp [hunk 1]
```diff
@@ -108,6 +107,11 @@ ActionDetection::ActionDetection(const ActionDetectorConfig& config)
                 const auto anchor_dims = outputInfo[glob_anchor_name]->getDims();
                 anchor_height = new_network_ ? anchor_dims[2] : anchor_dims[1];
                 anchor_width = new_network_ ? anchor_dims[3] : anchor_dims[2];
+                decltype(anchor_dims.size()) action_dimention_idx = new_network_ ? 1 : 3;
```
Comment (IRDonch) [inline demos/smart_classroom_demo/src/action_detector.cpp]: Using `decltype` for this is a bit silly. You know it's going to be `size_t`.

Comment (Wovchena) [reply demos/smart_classroom_demo/src/action_detector.cpp]: replaced

File: demos/smart_classroom_demo/src/action_detector.cpp [hunk 2]
```diff
@@ -108,6 +107,11 @@ ActionDetection::ActionDetection(const ActionDetectorConfig& config)
                 const auto anchor_dims = outputInfo[glob_anchor_name]->getDims();
                 anchor_height = new_network_ ? anchor_dims[2] : anchor_dims[1];
                 anchor_width = new_network_ ? anchor_dims[3] : anchor_dims[2];
+                std::size_t action_dimention_idx = new_network_ ? 1 : 3;
+                if (anchor_dims[action_dimention_idx] != config_.num_action_classes) {
+                    throw std::logic_error("The number of specified actions and the number of actions predicted by "
+                        "the Person/Action Detection Retail model must match");
```
Comment (IRDonch) [inline demos/smart_classroom_demo/src/action_detector.cpp]: Retail?

File: demos/tests/cases.py [hunk 3]
```diff
@@ -189,10 +189,15 @@ def device_cases(*args):
...
-                single_option_cases('-m_act',
-                    ModelArg('person-detection-action-recognition-0005'),
-                    ModelArg('person-detection-action-recognition-0006'),
-                    ModelArg('person-detection-action-recognition-teacher-0002')),
+                [
+                    TestCase(options={'-m_act': ModelArg('person-detection-action-recognition-0005')}),
+                    TestCase(options={'-m_act': ModelArg('person-detection-action-recognition-0006'),
+                        '-student_ac': 'sitting,writing,raising_hand,standing,turned_around,lie_on_the_desk'}),
+                    # person-detection-action-recognition-teacher-0002 is supposed to be provided with -teacher_id, but
+                    # this would require providing a gallery file with -fg key. Unless -teqcher_id is provided
```
Comment (IRDonch) [inline demos/tests/cases.py]: teqcher -> teacher

File: demos/smart_classroom_demo/src/action_detector.cpp (hunk 1 above)
Comment (Wovchena) [reply demos/smart_classroom_demo/src/action_detector.cpp]: replaced

File: demos/smart_classroom_demo/src/action_detector.cpp (hunk 2 above)
Comment (Wovchena) [reply demos/smart_classroom_demo/src/action_detector.cpp]: That is how the demo refers to the models

File: demos/tests/cases.py (hunk 3 above)
Comment (Wovchena) [reply demos/tests/cases.py]: updated

**Pull request merged**